server/src/main/java/eu/fasten/server/plugins/kafka/FastenKafkaPlugin.java

```diff
@@ -209,7 +213,13 @@ public void handleConsuming() {
             doCommitSync(KafkaRecordKind.PRIORITY);
         }
 
-        if (!normTopics.isEmpty() && !hasConsumedPriorityRecord) {
+        if (!normTopics.isEmpty()) {
+            // Pause the normal consumer if there are priority records to process.
+            if (hasConsumedPriorityRecord) {
+                connNorm.pause(connNorm.assignment());
+            } else {
+                connNorm.resume(connNorm.assignment());
+            }
```

Comment:
Couldn't we just introduce an artificial "heartbeat" method like:



    private static void sendHeartBeat(KafkaConsumer<String, String> con) {
    	// see https://stackoverflow.com/a/43722731
    	con.pause(con.assignment());
    	con.poll(Duration.ZERO);
    	con.resume(con.assignment());
    }

And call this in the beginning of the poll loop? These three calls should be cheap, but refresh the timeout counter. I would prefer this solution over a complication of the already complex logic in the rest of the method...